src/core/Tools.cpp

```diff
@@ -240,4 +246,110 @@ void setupSearchPaths()
 #endif
 }
 
+//
+// Prevent memory dumps without admin privileges.
+// MiniDumpWriteDump function requires PROCESS_QUERY_INFORMATION and PROCESS_VM_READ
+// see: https://msdn.microsoft.com/en-us/library/windows/desktop/ms680360%28v=vs.85%29.aspx
+//
```

Comment:
I just tested it on Windows 10 and it doesn't seem to work. I opened the task manager, right-clicked KeePassXC and chose "Create dump file". After a short while it said "The file has been successfully created."